docs/operation/Recover_From_Etcd_Permanent_Quorum_Loss.md

```diff
@@ -0,0 +1,82 @@
+
+   ```
+     Volumes:
+      etcd-config-file:
+       Type:      ConfigMap (a volume populated by a ConfigMap)
+       Name:      etcd-bootstrap-4785b0
+       Optional:  false
+   ```
+
+      Alternatively, the related configmap name can be obtained by executing following command as well:
+   `kubectl get sts etcd-main -o jsonpath='{.spec.template.spec.volumes[?(@.name=="etcd-config-file")].configMap.name}'`
+
+   3. Scale down the `etcd-main` statefulset replicas to `0`
+
+      `kubectl scale sts etcd-main --replicas=0`
+   4. The PVCs will look like the following on listing them with the command `kubectl get pvc` :
+      
+      ```
+      main-etcd-etcd-main-0        Bound    pv-shoot--garden--aws-ha-dcb51848-49fa-4501-b2f2-f8d8f1fad111   80Gi       RWO            gardener.cloud-fast   13d
+      main-etcd-etcd-main-1        Bound    pv-shoot--garden--aws-ha-b4751b28-c06e-41b7-b08c-6486e03090dd   80Gi       RWO            gardener.cloud-fast   13d
+      main-etcd-etcd-main-2        Bound    pv-shoot--garden--aws-ha-ff17323b-d62e-4d5e-a742-9de823621490   80Gi       RWO            gardener.cloud-fast   13d
+      ```
+      Delete all PVCs that are attached to `etcd-main` cluster.
+
+      `kubectl delete pvc -l instance=etcd-main`
+   5. Edit the `etcd-main` cluster's configmap (ex: `etcd-bootstrap-4785b0`) as follows:
+      
+      Find the `initial-cluster` field in the configmap. It will look like the following:
+      ```
+      # Initial cluster
+        initial-cluster: etcd-main-0=https://etcd-main-0.etcd-main-peer.default.svc:2380,etcd-main-1=https://etcd-main-1.etcd-main-peer.default.svc:2380,etcd-main-2=https://etcd-main-2.etcd-main-peer.default.svc:2380
+      ```
+
+      Change the `initial-cluster` field to have only one member (`etcd-main-0`) in the string. It should now look like this:
+
+      ```
+      # Initial cluster
+        initial-cluster: etcd-main-0=https://etcd-main-0.etcd-main-peer.default.svc:2380
+      ```
+   6. Scale up the `etcd-main` statefulset replicas to `1`
+
+      `kubectl scale sts etcd-main --replicas=1`
+   7. Wait for the single-member etcd cluster to be completely ready.
+
+      `kubectl get pods etcd-main-0` will give the following output when ready:
+      ```
+      NAME          READY   STATUS    RESTARTS   AGE
+      etcd-main-0   2/2     Running   0          1m
+      ```
+   8. Add the following annotation to the `etcd-main` statefulset `kubectl annotate sts etcd-main gardener.cloud/scaled-to-multi-node="true"`
+   9. Remove the following annotation from the `Etcd` resource `etcd-main`: `kubectl annotate etcd etcd-main druid.gardener.cloud/ignore-reconciliation-`
+   10. Finally add the following annotation to the `Etcd` resource `etcd-main`: `kubectl annotate etcd etcd-main gardener.cloud/operation="reconcile"`
+   11. Verify that the etcd cluster is formed correctly.
```

Comment:
I'm not getting how you are verifying whether the etcd cluster formed correctly or not.

Reply:
I made some changes for this. Please check